Sources/SwiftProtobuf/ExtensionFieldValueSet.swift

```diff
@@ -15,75 +15,67 @@
 
-  public static func ==(lhs: ExtensionFieldValueSet,
-                        rhs: ExtensionFieldValueSet) -> Bool {
-    guard lhs.values.count == rhs.values.count else {
-      return false
-    }
-    for (index, l) in lhs.values {
-      if let r = rhs.values[index] {
-        if type(of: l) != type(of: r) {
-          return false
-        }
-        if !l.isEqual(other: r) {
-          return false
-        }
-      } else {
-        return false
+      public static func == (
+            lhs: ExtensionFieldValueSet, rhs: ExtensionFieldValueSet
+      ) -> Bool {
+            guard lhs.values.count == rhs.values.count else { return false }
+            for (index, l) in lhs.values {
+                  if let r = rhs.values[index] {
+                        if type(of: l) != type(of: r) { return false }
+                        if !l.isEqual(other: r) { return false }
+                  }
+                  else { return false }
+            }
+            return true
       }
-    }
-    return true
-  }
 
-  public init() {}
+      public init() {}
 
-#if swift(>=4.2)
-  public func hash(into hasher: inout Hasher) {
-    // AnyExtensionField is not Hashable, and the Self constraint that would
-    // add breaks some of the uses of it; so the only choice is to manually
-    // mix things in. However, one must remember to do things in an order
-    // independent manner.
-    var hash = 16777619
-    for (fieldNumber, v) in values {
-      var localHasher = hasher
-      localHasher.combine(fieldNumber)
-      v.hash(into: &localHasher)
-      hash = hash &+ localHasher.finalize()
-    }
-    hasher.combine(hash)
-  }
-#else  // swift(>=4.2)
-  public var hashValue: Int {
-    var hash = 16777619
-    for (fieldNumber, v) in values {
-      // Note: This calculation cannot depend on the order of the items.
-      hash = hash &+ fieldNumber &+ v.hashValue
-    }
-    return hash
-  }
-#endif  // swift(>=4.2)
+      #if swift(>=4.2)
+        public func hash(into hasher: inout Hasher) {
+          // AnyExtensionField is not Hashable, and the Self constraint that would
+          // add breaks some of the uses of it; so the only choice is to manually
+          // mix things in. However, one must remember to do things in an order
+          // independent manner.
+          var hash = 16_777_619
+          for (fieldNumber, v) in values {
+            var localHasher = hasher
+            localHasher.combine(fieldNumber)
+            v.hash(into: &localHasher)
+            hash = hash &+ localHasher.finalize()
+          }
+          hasher.combine(hash)
+        }
+      #else  // swift(>=4.2)
+        public var hashValue: Int {
+          var hash = 16_777_619
+          for (fieldNumber, v) in values {
+            // Note: This calculation cannot depend on the order of the items.
+            hash = hash &+ fieldNumber &+ v.hashValue
+          }
+          return hash
+        }
+      #endif
```

Comment:
The comment on this `#endif` seems to have gotten lots, but the on on the `#else` above survived.

Reply:
@dabelknap Did I break this, or was this a known issue that we just hadn't fixed yet?

Reply:
This is a pre-existing issue. The problem is that the entire `#if/#endif` block is an UnknownStmt, and being printed verbatim. I don't think we're currently checking for trailing comments on verbatim tokens. Since the missing comment belongs to the `public` keyword, we're missing it.

Reply:
This is now fixed with PR https://github.com/google/swift/pull/213. Ordinarily, `#if` blocks are parsed by SwiftSyntax correctly. But, for some reason nesting it inside of a `struct` causes it to be parsed as an `UnknownStmt`.